Remove global reference in sources

soelen opened the pull request:
I like to develop in an esm based enviroment, unfortunatly the `three-pathfinding.module.js` build is not importing three in any shape or form, which is quite an issue for people which do not want to reference three globally :)

I decided to give it a try and tried to adjust things in three.js style, meaning to also use the import statments in the sources and to distribute with `rolllup` later on. While I didn't know zero configuration bundles like `microbundle` existed ( which is an awesome concept! ), I tended to switch to rollup in order to have more control how things will be built. I think this is especially important because some of the builds come out with some missconfigurations, at least IMHO:

- Why is the default dist (`dist/three-pathfinding.js`) a commonjs module ( which also requires THREE globally ), shouldnt this be, idk, an `IIFE` for example?
- Why is the minified version bigger than the default version? Why are both versions minified?
- For that matter, why is everthing minified? For the sake good practices the module build shouldn't be minified as it is not in the module version of [three](https://raw.githubusercontent.com/mrdoob/three.js/dev/build/three.module.js)
- Etc.

But I am not sure if this goes with the your intentions where this repository should go to. Please let me know what you think of these changes!

I also didnt tested yet / did not write any tests if every build is working as supposed to because I would like to hear your oppinion first before I put more work into it!

Cheers!

donmccurdy commented:
Hi @soelen — the changes in the `src/` folder, and making `three` an external dependency, look good to me. However, I'd prefer to continue using microbundle. I have a number of repositories to maintain, and using a single (mostly config-free) bundler makes this easier for me. I think that `--globals three=THREE` in the microbundle command might be enough, or `--external three` in addition if needed.

On your specific questions:


> Why is the default dist (dist/three-pathfinding.js) a commonjs module ...

It's expected for the `main` entry of `package.json` to be a CommonJS-compatible file. In this case it's actually UMD, which can be used like an IIFE but is still CommonJS-compatible.

> Why is the minified version bigger than the default version? Why are both versions minified?

The `three-pathfinding.min.js` is two years old, I think I must have forgotten to delete it after changing the build step to something that didn't create this file anymore. I'll get rid of that. :)

> For that matter, why is everthing minified? For the sake good practices the module build shouldn't be minified...

The remaining files are minified because I've otherwise left Microbundle's settings alone. Because it also generates sourcemaps I don't find this to be a problem, but there's some prior discussion in https://github.com/developit/microbundle/issues/66.

If you're interested in making these changes, please include only `src/` and `package.json` in the PR. The rest of the changes will get applied when I bundle the next release. Thanks!

soelen commented:
Hey @donmccurdy, thank you for your response. I adjusted the PR and included `src/` ,`package.json` and `.gitingore`, I hope this is ok.

Tried to read a bit about microbundle and how to bundle things with it, changed `--target browser` to `--target web` ( I think the options for this parameters has been updated, there is no target browser according to the docs ) and tried to play with the `globals` and `external` but I am actually not sure if these parameters has been set corrently.

Please let me know if I am on the right track!

donmccurdy commented:
Thanks! 

I did revert `.gitignore`, and seemed to need both `--external` and `--globals`, but this works!
